roles/sudoers/tasks/main.yml

```diff
@@ -17,17 +17,18 @@
     path: "{{ item }}"
     state: absent
   with_items: "{{ ansible_managed_sudoers.files | map(attribute='path') | list }}"
-  when: item | regex_replace('.*ansible-managed-','') not in regular_users | default([]) | selectattr('sudoers', 'defined') | map(attribute='user') | list
+  when: item | regex_replace('.*ansible-managed-','') not in sudoers | default('') | selectattr('name', 'defined') | map(attribute='name') | list
   become: true
 
 - name: 'Allow passwordless sudo to functional accounts.'
   ansible.builtin.template:
     src: 'templates/92-ansible-managed'
-    dest: "/etc/sudoers.d/92-ansible-managed-{{ item.user }}"
+    dest: "/etc/sudoers.d/92-ansible-managed-{{ item.name }}"
     owner: 'root'
     group: 'root'
     mode: '0440'
+    validate: '/usr/sbin/visudo -cf %s'
   with_items:
-    - "{{ regular_users | default([]) | selectattr('sudoers', 'defined') | list }}"
+    - "{{ sudoers | selectattr('name', 'defined') }}"
```

Comment:
```suggestion
    - "{{ sudoers | default([]) }}"
```

Reply:
- If `sudoers` is not defined in a stack's `group_vars`, this role should not fail, but skip this task.
- if `sudoers` is defined, but one of the items is missing a `name` attribute, then this task should fail as this is a bug.